src/ChatPage.cc

```diff
@@ -139,6 +142,10 @@ ChatPage::ChatPage(QSharedPointer<MatrixClient> client, QWidget *parent)
 
                 typingDisplay_->setUsers(users);
         });
+        connect(room_list_, &RoomList::roomChanged, this, [=]() {
+                typingRefresher_->stop();
+                text_input_->stopTyping();
```

Comment:
If I'm not mistaken `stopTyping` will emit the `stoppedTyping` event which is also handled here. Maybe we can just call `MatrixClient::sendTypingStoppedNotifications` here directly like the `stoppedTyping` handler does. The timer on the text input will expire anyway.